### PR TITLE
Add "[" and "]" operators

### PR DESCRIPTION
**Pull request by HendrikRoehm**

### CODE REVIEW

**Comment by HendrikRoehm** [comment]
Includes an interface refactoring

File: lsystems/svg.py [hunk 1]
```diff
@@ -2,32 +2,28 @@
...
+    x: float
+    y: float
+    dir: float
+
+    def __init__(self, x: float, y: float, dir: float):
+        self.x = x
+        self.y = y
+        self.dir = dir
+
+    def __eq__(self, other):
```
Comment (mschlund) [inline lsystems/svg.py:15]: perhaps we {c|sh}ould use math.isclose?
https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python

File: lsystems/turtle.py [hunk 2]
```diff
@@ -70,6 +65,9 @@ def __init__(self, angle, stride, size, width=3):
             "L": Rotation(angle=angle),
             "R": Rotation(angle=-angle),
             ")": Arc(angle=angle, rx=stride, ry=stride),
-            "(": Arc(angle=-angle, rx=stride, ry=stride)
+            "(": Arc(angle=-angle, rx=stride, ry=stride),
+            "[": PushPosition(stack=stack),
+            "]": PopPosition(stack=stack)
+            # TODO: we can extend this to other brackets like "{" with giving PushPosition an argument
```
Comment (mschlund) [inline lsystems/turtle.py]: "TODO" can be deleted -- I'll create an issue instead (is a bit cleaner than having TODOs scattered in the code ;))

File: test/test_svg.py [hunk 3]
```diff
@@ -1,47 +1,62 @@
...
+        self.assertEqual(end_cursor, Cursor(x=11, y=30, dir=90), f"Falscher Cursor: {end_cursor}")
 
     def test_rotation(self):
         line = Rotation(angle=122)
-        segment = line.generate(previous_segment)
+        start_cursor = Cursor(x=11, y=10, dir=90)
+
+        segment, end_cursor = line.generate(start_cursor)
+
+        self.assertEqual(end_cursor, Cursor(x=11, y=10, dir=90-122), f"Falscher Cursor: {end_cursor}")
```
Comment (mschlund) [inline test/test_svg.py]: error-messages should maybe be in English (also below) :)

**Comment by mschlund** [comment]
Very nice :) Maybe we can add the "FractalPlant" l-system as a curve to demonstrate the effects?
(https://en.wikipedia.org/wiki/L-system#Example_7:_Fractal_plant)

File: lsystems/turtle.py (hunk 2 above)
Comment (HendrikRoehm) [reply lsystems/turtle.py]: Actually, with the current solution, this is already implemented. Just use another variable like stack2 and add to more symbols with PushPosition and PopPosition using this other stack.

File: lsystems/svg.py (hunk 1 above)
Comment (HendrikRoehm) [reply lsystems/svg.py:15]: done

File: test/test_svg.py (hunk 3 above)
Comment (HendrikRoehm) [reply test/test_svg.py]: done

**Comment by HendrikRoehm** [comment]
@mschlund : Changes are integrated

**Comment by mschlund** [comment]
lgtm

**Pull request merged**